app/lib/search/backend.dart

```diff
@@ -138,6 +147,71 @@ class SearchBackend {
   }
 }
 
+class PopularityStorage {
+  final Storage storage;
+  final Bucket bucket;
+
+  final Map<String, double> _values = {};
+
+  PopularityStorage(this.storage, this.bucket);
+
+  double operator [](String package) => _values[package];
+
+  Future init() async {
+    await fetch();
+    new Timer.periodic(const Duration(days: 1), (_) {
+      fetch();
+    });
+  }
+
+  Future fetch() async {
+    _logger.info('Updating popularity storage...');
+    final List<BucketEntry> list = await bucket.list().toList();
+    final List<String> names = list
+        .where((entry) => entry.isObject)
+        .map((entry) => entry.name)
+        .toList();
+    if (names.isEmpty) {
+      _logger.warning('No popularity data found!');
+      return;
+    }
+    // Try to load the available snapshots in reverse order (latest first).
+    names.sort();
+    for (String selected in names.reversed) {
+      _logger.info('Loading popularity data: $selected');
+      try {
+        final String json =
+            await bucket.read(selected).transform(UTF8.decoder).join();
```

Comment:
Also we should consider zipping the file, usually it gives 10x size reductions and the cost of unzipping is low.